cpp/src/io/parquet/page_decode.cuh

```diff
@@ -924,7 +924,7 @@ inline __device__ uint32_t InitLevelSection(page_state_s* s,
 
   auto start = cur;
 
-  auto init_rle = [s, lvl, end, level_bits](uint8_t const* cur, uint8_t const* end) {
+  auto init_rle = [s, lvl, level_bits](uint8_t const* cur, uint8_t const* end) {
```

Comment:
The behavior when a captured variable and a parameter have the same name is interesting. Thanks to godbolt, I found that all options are possible, with different compilers -  captured variable value is used, or parameter value is used, or compilation fails. Fun!

Reply:
Wow that's interesting. We were in the same situation before, and the code runs differently on different architectures (due to compiled into different archs?).

Reply:
Thank you clangd! :smile:

Reply:
I think nvcc stayed consistent in prioritizing the parameter over the capture. But I found online that other compilers disagree.

Reply:
We found that the capture overshadows the previous variable on H100+V100 while on other GPUs it doesn't.